### DISCUSSION TITLE
Allow to process all sysex messages

### DISCUSSION THREAD

**Discussion by trolley813** [FluidSynth/fluidsynth#1206] [Ideas]
According to the [docs](https://www.fluidsynth.org/api/fluidsettings.xml#synth.device-id), FluidSynth processes only messages destined for this ID (except for broadcast messages with ID=127, which are processed in any case). But for users which "just play MIDI files" (as said in #1092, precisely [here](https://github.com/FluidSynth/fluidsynth/issues/1092#issuecomment-1330158403) - although it's almost unrelated to the problem) this may be not the best case, especially when dealing with XG, GS etc. files, since one does have to know exactly which device IDs are used in the file.

My suggestion is to allow processing any sysex messages regardless of the device ID, e.g. by setting `synth.device-id` to "useless" value of 127 (which does not make actual sense, and the available range is 0 to 126 anyway).

**Comment by derselbst** [comment]
> My suggestion is to allow processing any sysex messages regardless of the device ID, e.g. by setting synth.device-id to "useless" value of 127 (which does not make actual sense, and the available range is 0 to 126 anyway).

Yeah, ok. Setting it to the broadcast ID sounds reasonable to process all SysEx. I welcome your PR.

**Reply by trolley813** [reply]
>  I welcome your PR.

Here it is. #1209 (maybe needs some additional polish).
